### PR TITLE
StateChecker: Don't save hash debug traces

### PR DESCRIPTION
**Pull request by Dillon-Brown**
## Description

An alternative to #7087 to disable storing hash debug traces while running the tests but without removing the functionality altogether.

## How To Test This

Run the unit tests.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Comment by lazka** [comment]
> but without removing the functionality altogether.

Note that it isn't used at all, so there is no removed functionality. Unless I'm missing something.

**Comment by Dillon-Brown** [comment]
@lazka It's only used during development to get a traceback for failing tests.

**Comment by lazka** [comment]
> @lazka It's only used during development to get a traceback for failing tests.

ok, do you mean you change the code of the statechecker during development to print something?

**Comment by Dillon-Brown** [comment]
@lazka Yeah, I haven't personally used these functions myself but it's my understanding they were added by @gymad to make debugging failing tests easier during development by collecting the stack trace.

**Comment by lazka** [comment]
Ah, ok, I wasn't aware of that.

**Comment by connorshea** [comment]
This doesn't change it to be disabled by default locally, does it? I think that's important as well.

File: .travis.yml [hunk 1]
```diff
@@ -86,7 +86,9 @@ script:
   - ./build/push_output.sh
   
   # Run the unit tests
-  # - ./vendor/bin/codecept run unit --steps -f
+  - sudo chmod -R 777 .
+  - echo "\$sugar_config['state_checker']['save_traces'] = false;
```
Comment (connorshea) [inline .travis.yml]: This doesn't actually do anything. Needs this:
```suggestion
  - echo "\$sugar_config['state_checker']['save_traces'] = false;" >> config_override.php
```
Same for the line below it.

**Comment by connorshea** [comment]
This doesn't seem to decrease the time it takes to run the test suite (e.g. acceptance tests with PHP 7.2 on this branch takes 13.84 minutes vs. 12.67 minutes on #7087), though I'm not sure if the time taken in #7087 can be attributed to the suite being run 3 months ago vs. now. For comparison, the acceptance suite on PHP 7.2 took 13.53 minutes in #7339.

@lazka could you rebase #7087 to see if it still has a time improvement over the current test runs?

**Comment by lazka** [comment]
> @lazka could you rebase #7087 to see if it still has a time improvement over the current test runs?

done. Note that my main motivation was the  memory usage as running the tests still takes >30GB for me.

**Comment by connorshea** [comment]
@lazka time to invest in one of those new Mac Pros with 1.5TB of RAM, I guess.

**Comment by lazka** [comment]
* before: Time: 3.27 minutes, Memory: 714.50MB
* after: Time: 3.06 minutes, Memory: 208.50MB

**Comment by connorshea** [comment]
@lazka is that for this PR or #7087?

**Comment by lazka** [comment]
my PR, but this one is ~ the same: "Time: 3.1 minutes, Memory: 208.50MB".

**Comment by cameronblaikie** [comment]
Assessed :+1:

**Pull request merged**